File: client-netcore/client-netcore/Giis.Samples.Openapi.Test.Netcore/TestPetApiClient.cs
using Giis.Samples.Openapi.Api;
using Giis.Samples.Openapi.Client;
using Giis.Samples.Openapi.Model;
using NUnit.Framework;
using NUnit.Framework.Legacy;
using System.Collections.Generic;

namespace Giis.Samples.Openapi.Test.Netcore
{
    public class Tests
    {
        private readonly PetsApi api = new PetsApi("http://localhost:8080");

        [SetUp]
        public void SetUp()
        {
            //restaura el contenido inicial de los pets para que los tests sean independientes
            api.ResetPets();
        }

        [Test]
        public void TestGetAllPets()
        {
            List<Pet> pets = api.ListPets(10);
            ClassicAssert.AreEqual(2, pets.Count);
            ClassicAssert.AreEqual("1", pets[0].Id.ToString());
            ClassicAssert.AreEqual("cat", pets[0].Name);
            ClassicAssert.AreEqual("2", pets[1].Id.ToString());
            ClassicAssert.AreEqual("dog", pets[1].Name);
        }
        [Test]
        public void TestGetExistingPet()
        {
            Pet pet = api.ShowPetById("2");
            ClassicAssert.AreEqual("2", pet.Id.ToString());
            ClassicAssert.AreEqual("dog", pet.Name);
        }
        [Test]
        public void TestPostAndGet()
        {
            //el id lo asigna el servidor y se devuelve en el pet creado
            Pet newPet = api.CreatePetQuery("mouse");
            ClassicAssert.AreEqual("3", newPet.Id.ToString());
            ClassicAssert.AreEqual("mouse", newPet.Name);
            Pet pet = api.ShowPetById("3");
            ClassicAssert.AreEqual("3", pet.Id.ToString());
            ClassicAssert.AreEqual("mouse", pet.Name);
        }
        [Test]
        public void TestGetNotExistingPet()
        {
            //If spring server returns null, spring client receives null, but apache http client ant netcore does not
            try
            {
                api.ShowPetById("0");
                ClassicAssert.Fail("should return excepton code 404 not found");
            }
            catch (ApiException e)
            {
                ClassicAssert.AreEqual(404, e.ErrorCode);
            }
        }

    }
}

File: client-resttemplate/src/test/java/giis/samples/openapi/test/resttemplate/TestPetApiClient.java
package giis.samples.openapi.test.resttemplate;

import org.junit.Test;
import org.junit.rules.TestName;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.client.HttpClientErrorException;

import giis.samples.openapi.api.PetsApi;
import giis.samples.openapi.invoker.ApiClient;
import giis.samples.openapi.model.Pet;
import giis.samples.openapi.model.Pets;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

import org.junit.Before;
import org.junit.Rule;

public class TestPetApiClient {
	private final static Logger log=LoggerFactory.getLogger(TestPetApiClient.class);
    private final PetsApi api = new PetsApi(new ApiClient().setBasePath("http://localhost:8080"));
    @Rule public TestName name = new TestName();

    @Before
    public void setUp() {
    	log.info(name.getMethodName());
    }
    @Test
    public void testGetAllPets() {
		Pets pets = api.listPets(10);
		assertTrue(2<=pets.size()); //puede haber mas de dos si se ejecuta antes el metodo que crea pets
		assertEquals("1", pets.get(0).getId().toString());
		assertEquals("cat", pets.get(0).getName());
		assertEquals("2", pets.get(1).getId().toString());
		assertEquals("dog", pets.get(1).getName());
    }
    @Test
    public void testGetExistingPet() {
        Pet pet = api.showPetById("2");
		assertEquals("2", pet.getId().toString());
		assertEquals("dog", pet.getName());
    }
    @Test
    public void testPostAndGet() {
    	//para que sea repetible el id y name se calculan como el siguiente valor de la secuencia de ids existentes 
    	//(pues no se hace un reset de los pets en setup)
    	int newId=api.listPets(10).size()+1;
    	String newName="mouse"+newId;
        api.createPetQuery(newId, newName);
		Pet pet = api.showPetById(String.valueOf(newId));
		assertEquals(String.valueOf(newId), pet.getId().toString());
		assertEquals(newName, pet.getName());
    }
    @Test
    public void testGetNotExistingPet() {
        //If spring server returns null, spring client receives null, but apache http client ant netcore does not
    	//Most clients return ApiException, but spring client returns its own exception
    	try {
    		api.showPetById("0");
           	fail("should return excepton code 404 not found");
    	} catch (HttpClientErrorException e) {
    		assertEquals("404 NOT_FOUND", e.getStatusCode().toString());
    	}
    }

}
